### PR TITLE
rustdoc: use unicode-aware checks for redundant explicit link fastpath

### LINKED ISSUE #115062: rustdoc collect intralinks ICE
### Code

```Rust
cargo doc --no-deps twilight-lavalink
```
On [twilight-lavalink](https://crates.io/crates/twilight-lavalink) (0.15.1 and latest `main`)

### Meta

Builds on
`rustc --version --verbose`:
```
rustc 1.73.0-nightly (6ef7d16be 2023-08-19)
binary: rustc
commit-hash: 6ef7d16be0fb9d6ecf300c27990f4bff49d22d46
commit-date: 2023-08-19
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 17.0.0
```

ICE on
```
rustc 1.74.0-nightly (5c6a7e71c 2023-08-20)
binary: rustc
commit-hash: 5c6a7e71cd66705c31c9af94077901a220f0870c
commit-date: 2023-08-20
host: x86_64-unknown-linux-gnu
release: 1.74.0-nightly
LLVM version: 17.0.0
```

### Error output

```
thread 'rustc' panicked at src/librustdoc/passes/collect_intra_doc_links.rs:1446:33:
byte index 8 is not a char boundary; it is inside '’' (bytes 6..9) of `client’s process method`
```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
stack backtrace:
   0:     0x7f59d3b630fc - std::backtrace_rs::backtrace::libunwind::trace::h68b3f03db065435f
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f59d3b630fc - std::backtrace_rs::backtrace::trace_unsynchronized::hcc29d957a519c467
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f59d3b630fc - std::sys_common::backtrace::_print_fmt::h51e5567c7446b80e
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f59d3b630fc - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h54890bb4d0b3c330
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f59d3bc93bc - core::fmt::rt::Argument::fmt::heae7cd8ccc900aeb
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/fmt/rt.rs:138:9
   5:     0x7f59d3bc93bc - core::fmt::write::hc761729f1163b685
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/fmt/mod.rs:1094:21
   6:     0x7f59d3b55d0e - std::io::Write::write_fmt::hfd968165f9447824
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/io/mod.rs:1714:15
   7:     0x7f59d3b62ee4 - std::sys_common::backtrace::_print::h1b2adaff5c947b3b
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f59d3b62ee4 - std::sys_common::backtrace::print::hf6a9b81f93e62890
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f59d3b65fda - std::panicking::panic_hook_with_disk_dump::{{closure}}::h663629ab1a73f13e
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:278:22
  10:     0x7f59d3b65c89 - std::panicking::panic_hook_with_disk_dump::h7dc7628c0cba33b1
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:305:9
  11:     0x7f59d6d12489 - <rustc_driver_impl[4490db7317d016ab]::install_ice_hook::{closure#0} as core[84f2f5510e338f25]::ops::function::FnOnce<(&core[84f2f5510e338f25]::panic::panic_info::PanicInfo,)>>::call_once::{shim:vtable#0}
  12:     0x7f59d3b66880 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h47f6ec65be0f4981
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/alloc/src/boxed.rs:2021:9
  13:     0x7f59d3b66880 - std::panicking::rust_panic_with_hook::hdd1521f6ef2543b7
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:733:13
  14:     0x7f59d3b66607 - std::panicking::begin_panic_handler::{{closure}}::h7c4ede115dee30b0
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:621:13
  15:     0x7f59d3b63626 - std::sys_common::backtrace::__rust_end_short_backtrace::habd7cadfbd319a50
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys_common/backtrace.rs:170:18
  16:     0x7f59d3b66352 - rust_begin_unwind
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:617:5
  17:     0x7f59d3bc57c3 - core::panicking::panic_fmt::hb85e049a4301ecd3
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/panicking.rs:67:14
  18:     0x7f59d3bce41a - core::str::slice_error_fail_rt::hb65a34b1fae3fa38
  19:     0x7f59d3bce227 - core::str::slice_error_fail::h3b10f159df42c419
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/str/mod.rs:87:9
  20:     0x55ea04eb5445 - <rustdoc[3c46f6e6ced9b5c4]::passes::collect_intra_doc_links::LinkCollector>::resolve_links
  21:     0x55ea04ea207d - rustdoc[3c46f6e6ced9b5c4]::passes::collect_intra_doc_links::collect_intra_doc_links
  22:     0x55ea04edbaee - <rustc_session[f827a11bbd15a9f6]::session::Session>::time::<(rustdoc[3c46f6e6ced9b5c4]::clean::types::Crate, rustdoc[3c46f6e6ced9b5c4]::config::RenderOptions, rustdoc[3c46f6e6ced9b5c4]::formats::cache::Cache), rustdoc[3c46f6e6ced9b5c4]::main_args::{closure#1}::{closure#0}::{closure#0}::{closure#0}>
  23:     0x55ea04dcdcf2 - <rustc_middle[ae9f4c9b3fbb3490]::ty::context::GlobalCtxt>::enter::<rustdoc[3c46f6e6ced9b5c4]::main_args::{closure#1}::{closure#0}::{closure#0}, core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>>
  24:     0x55ea04ee00fc - <scoped_tls[ff34cbc9fa0ab60c]::ScopedKey<rustc_span[e97895297a92aa57]::SessionGlobals>>::set::<rustc_interface[b2bba20ffff10a85]::interface::run_compiler<core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>, rustdoc[3c46f6e6ced9b5c4]::main_args::{closure#1}>::{closure#0}, core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>>
  25:     0x55ea04fcd629 - std[7240a06e7b8f268c]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[b2bba20ffff10a85]::util::run_in_thread_pool_with_globals<rustc_interface[b2bba20ffff10a85]::interface::run_compiler<core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>, rustdoc[3c46f6e6ced9b5c4]::main_args::{closure#1}>::{closure#0}, core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>>
  26:     0x55ea04f876f1 - <<std[7240a06e7b8f268c]::thread::Builder>::spawn_unchecked_<rustc_interface[b2bba20ffff10a85]::util::run_in_thread_pool_with_globals<rustc_interface[b2bba20ffff10a85]::interface::run_compiler<core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>, rustdoc[3c46f6e6ced9b5c4]::main_args::{closure#1}>::{closure#0}, core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[84f2f5510e338f25]::result::Result<(), rustc_span[e97895297a92aa57]::ErrorGuaranteed>>::{closure#1} as core[84f2f5510e338f25]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  27:     0x7f59d3b711e5 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h7bbddabb40181918
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/alloc/src/boxed.rs:2007:9
  28:     0x7f59d3b711e5 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h4e41f7977be0d993
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/alloc/src/boxed.rs:2007:9
  29:     0x7f59d3b711e5 - std::sys::unix::thread::Thread::new::thread_start::h6eb72cf9b874b24b
                               at /rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/sys/unix/thread.rs:108:17
  30:     0x7f59d368c9eb - <unknown>
  31:     0x7f59d3710dfc - <unknown>
  32:                0x0 - <unknown>


rustc version: 1.74.0-nightly (5c6a7e71c 2023-08-20)
platform: x86_64-unknown-linux-gnu

query stack during panic:
end of query stack
```

</p>
</details>

### LINKED ISSUE #115064: ICE in rustdoc `redundant_explicit_links` with non-ASCII link text
### Code

Reduced from the source code of `bumpalo 3.13.0`:

```Rust
/// [`…foo`] [`…bar`] [`Err`]
pub struct Broken {}
```

### Meta

`rustc --version --verbose`:
```
rustc 1.74.0-nightly (5c6a7e71c 2023-08-20)
binary: rustc
commit-hash: 5c6a7e71cd66705c31c9af94077901a220f0870c
commit-date: 2023-08-20
host: x86_64-apple-darwin
release: 1.74.0-nightly
LLVM version: 17.0.0
```

### Error output

```
thread 'rustc' panicked at src/librustdoc/passes/lint/redundant_explicit_links.rs:105:38:
byte index 2 is not a char boundary; it is inside '…' (bytes 1..4) of ``…foo``
```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
thread 'rustc' panicked at src/librustdoc/passes/lint/redundant_explicit_links.rs:105:38:
byte index 2 is not a char boundary; it is inside '…' (bytes 1..4) of ``…foo``
stack backtrace:
   0:        0x10627bf0b - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h0a889342af22c001
   1:        0x1062dc77b - core::fmt::write::h2a9552489c640d30
   2:        0x10627098e - std::io::Write::write_fmt::h957f1bc49a89c995
   3:        0x10627bce9 - std::sys_common::backtrace::print::hd3ca24f2b09e2d6b
   4:        0x10627f3c8 - std::panicking::panic_hook_with_disk_dump::{{closure}}::hfdbd4926b1318eff
   5:        0x10627f090 - std::panicking::panic_hook_with_disk_dump::hff36df8bfb145345
   6:        0x114f89181 - rustc_driver_impl[271211c656e69f7c]::install_ice_hook::{closure#0}
   7:        0x10627fc10 - std::panicking::rust_panic_with_hook::h3d225d93354d6850
   8:        0x10627f9b3 - std::panicking::begin_panic_handler::{{closure}}::hfb6f37b7f59b8143
   9:        0x10627c3e9 - std::sys_common::backtrace::__rust_end_short_backtrace::h5cb50ae720fa1297
  10:        0x10627f6dd - _rust_begin_unwind
  11:        0x106308023 - core::panicking::panic_fmt::hda3b11d26a2996a0
  12:        0x1062e1595 - core::str::slice_error_fail_rt::hb64c0d7c400011f0
  13:        0x1063087b9 - core::str::slice_error_fail::h4ada315b265e1d30
  14:        0x104865f69 - rustdoc[7e76fc70b3a238cf]::passes::lint::redundant_explicit_links::visit_item
  15:        0x10483b8d7 - <rustdoc[7e76fc70b3a238cf]::passes::lint::Linter as rustdoc[7e76fc70b3a238cf]::visit::DocVisitor>::visit_item
  16:        0x10483b9bc - <rustdoc[7e76fc70b3a238cf]::passes::lint::Linter as rustdoc[7e76fc70b3a238cf]::visit::DocVisitor>::visit_item
  17:        0x10483b3f8 - rustdoc[7e76fc70b3a238cf]::passes::lint::run_lints
  18:        0x10476b05a - <rustc_session[aab8e8868057613]::session::Session>::time::<(rustdoc[7e76fc70b3a238cf]::clean::types::Crate, rustdoc[7e76fc70b3a238cf]::config::RenderOptions, rustdoc[7e76fc70b3a238cf]::formats::cache::Cache), rustdoc[7e76fc70b3a238cf]::main_args::{closure#1}::{closure#0}::{closure#0}::{closure#0}>
  19:        0x10483e03b - <rustc_middle[110db10588286083]::ty::context::GlobalCtxt>::enter::<rustdoc[7e76fc70b3a238cf]::main_args::{closure#1}::{closure#0}::{closure#0}, core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>>
  20:        0x10477001c - <scoped_tls[14d6625b228cefa1]::ScopedKey<rustc_span[3fe5612a2015833b]::SessionGlobals>>::set::<rustc_interface[81a8130e43f77706]::interface::run_compiler<core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>, rustdoc[7e76fc70b3a238cf]::main_args::{closure#1}>::{closure#0}, core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>>
  21:        0x1049f9df3 - std[8dadf9e2e2548a40]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[81a8130e43f77706]::util::run_in_thread_pool_with_globals<rustc_interface[81a8130e43f77706]::interface::run_compiler<core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>, rustdoc[7e76fc70b3a238cf]::main_args::{closure#1}>::{closure#0}, core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>>
  22:        0x1049fa56e - <<std[8dadf9e2e2548a40]::thread::Builder>::spawn_unchecked_<rustc_interface[81a8130e43f77706]::util::run_in_thread_pool_with_globals<rustc_interface[81a8130e43f77706]::interface::run_compiler<core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>, rustdoc[7e76fc70b3a238cf]::main_args::{closure#1}>::{closure#0}, core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[cee64f789374d93]::result::Result<(), rustc_span[3fe5612a2015833b]::ErrorGuaranteed>>::{closure#1} as core[cee64f789374d93]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  23:        0x106289949 - std::sys::unix::thread::Thread::new::thread_start::h4626e4e0af234279
  24:     0x7ff817d4b4e1 - __pthread_start


rustc version: 1.74.0-nightly (5c6a7e71c 2023-08-20)
platform: x86_64-apple-darwin

query stack during panic:
end of query stack
```

</p>
</details>

### Other notes

Would have been introduced by #113167 since that PR added the lint.. cc @ChAoSUnItY 

@rustbot label -T-compiler +T-rustdoc

### LINKED ISSUE #115116: `cargo doc` => thread 'rustc' panicks with "byte index 2 is not a char boundary ..." on nightly
### Code

The error only appears on nightly.

steps to reproduce:

```bash
cargo init                    # or cargo init --lib
cargo add axum       # this package caused the error
cargo doc
```

`rustc --version --verbose`:
```
rustc 1.74.0-nightly (ef85656a1 2023-08-21)
binary: rustc
commit-hash: ef85656a10657ba5e4f7fe2931a4ca6293138d51
commit-date: 2023-08-21
host: x86_64-unknown-linux-gnu
release: 1.74.0-nightly
LLVM version: 17.0.0
```

`cargo version --verbose`
```
cargo 1.74.0-nightly (80eca0e58 2023-08-19)
release: 1.74.0-nightly
commit-hash: 80eca0e58fb2ff52c1e94fc191b55b37ed73e0e4
commit-date: 2023-08-19
host: x86_64-unknown-linux-gnu
libgit2: 1.6.4 (sys:0.17.2 vendored)
libcurl: 8.2.1-DEV (sys:0.4.65+curl-8.2.1 vendored ssl:OpenSSL/1.1.1u)
ssl: OpenSSL 1.1.1u  30 May 2023
os: Fedora 37.0.0 [64-bit]
```

### Error output

```
Documenting axum v0.6.20
thread 'rustc' panicked at src/librustdoc/passes/collect_intra_doc_links.rs:1446:33:
byte index 2 is not a char boundary; it is inside '“' (bytes 0..3) of `“extractors”`
stack backtrace:
   0: rust_begin_unwind
             at /rustc/ef85656a10657ba5e4f7fe2931a4ca6293138d51/library/std/src/panicking.rs:617:5
   1: core::panicking::panic_fmt
             at /rustc/ef85656a10657ba5e4f7fe2931a4ca6293138d51/library/core/src/panicking.rs:67:14
   2: core::str::slice_error_fail_rt
   3: core::str::slice_error_fail
             at /rustc/ef85656a10657ba5e4f7fe2931a4ca6293138d51/library/core/src/str/mod.rs:87:9
   4: <rustdoc::passes::collect_intra_doc_links::LinkCollector>::resolve_links
   5: rustdoc::passes::collect_intra_doc_links::collect_intra_doc_links
   6: <rustc_session::session::Session>::time::<(rustdoc::clean::types::Crate, rustdoc::config::RenderOptions, rustdoc::formats::cache::Cache), rustdoc::main_args::{closure#1}::{closure#0}::{closure#0}::{closure#0}>
   7: <rustc_middle::ty::context::GlobalCtxt>::enter::<rustdoc::main_args::{closure#1}::{closure#0}::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>
   8: <scoped_tls::ScopedKey<rustc_span::SessionGlobals>>::set::<rustc_interface::interface::run_compiler<core::result::Result<(), rustc_span::ErrorGuaranteed>, rustdoc::main_args::{closure#1}>::{closure#0}, core::result::Result<(), rustc_span::ErrorGuaranteed>>
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-rustdoc&template=ice.md

note: please attach the file at `/home/luk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-0.6.20/rustc-ice-2023-08-22T21:28:21.787458591Z-17768.txt` to your bug report

note: compiler flags: --crate-type lib

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
end of query stack
error: could not document `axum`

Caused by:
  process didn't exit successfully: `/home/luk/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/rustdoc --edition=2021 --crate-type lib --crate-name axum /home/luk/.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-0.6.20/src/lib.rs --cap-lints allow -o /home/luk/code/target/doc --cfg 'feature="default"' --cfg 'feature="form"' --cfg 'feature="http1"' --cfg 'feature="json"' --cfg 'feature="matched-path"' --cfg 'feature="original-uri"' --cfg 'feature="query"' --cfg 'feature="tokio"' --cfg 'feature="tower-log"' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=174 -C metadata=6a2b919d3caa327b -L dependency=/home/luk/code/target/debug/deps --extern async_trait=/home/luk/code/target/debug/deps/libasync_trait-09449879fe59171c.so --extern axum_core=/home/luk/code/target/debug/deps/libaxum_core-be3802c8acc62418.rmeta --extern bitflags=/home/luk/code/target/debug/deps/libbitflags-17eaae058c49a1b8.rmeta --extern bytes=/home/luk/code/target/debug/deps/libbytes-b51fc775aa196866.rmeta --extern futures_util=/home/luk/code/target/debug/deps/libfutures_util-181be8ad31767cdd.rmeta --extern http=/home/luk/code/target/debug/deps/libhttp-777d680c671caa21.rmeta --extern http_body=/home/luk/code/target/debug/deps/libhttp_body-3675ad811d3791f2.rmeta --extern hyper=/home/luk/code/target/debug/deps/libhyper-26f28c600bdc1ee2.rmeta --extern itoa=/home/luk/code/target/debug/deps/libitoa-94c3240fd22fdaed.rmeta --extern matchit=/home/luk/code/target/debug/deps/libmatchit-82c8078bb1278ab5.rmeta --extern memchr=/home/luk/code/target/debug/deps/libmemchr-3693a015842dadb3.rmeta --extern mime=/home/luk/code/target/debug/deps/libmime-d49ab82920264418.rmeta --extern percent_encoding=/home/luk/code/target/debug/deps/libpercent_encoding-82f182e7e48872ba.rmeta --extern pin_project_lite=/home/luk/code/target/debug/deps/libpin_project_lite-57b44a4b14a3bb39.rmeta --extern serde=/home/luk/code/target/debug/deps/libserde-0d5b7588f54bc67d.rmeta --extern serde_json=/home/luk/code/target/debug/deps/libserde_json-278584084946d9dc.rmeta --extern serde_path_to_error=/home/luk/code/target/debug/deps/libserde_path_to_error-39acfc136a848aca.rmeta --extern serde_urlencoded=/home/luk/code/target/debug/deps/libserde_urlencoded-e540d2bfed2238c6.rmeta --extern sync_wrapper=/home/luk/code/target/debug/deps/libsync_wrapper-1f9c28405a3b7050.rmeta --extern tokio=/home/luk/code/target/debug/deps/libtokio-6e8d9240e9f399fd.rmeta --extern tower=/home/luk/code/target/debug/deps/libtower-3058a136415171e6.rmeta --extern tower_layer=/home/luk/code/target/debug/deps/libtower_layer-e590ca43fcd1237b.rmeta --extern tower_service=/home/luk/code/target/debug/deps/libtower_service-6e838e2bbdb7ccf9.rmeta --crate-version 0.6.20 --cfg nightly_error_messages` (exit status: 101)
```
[rustc-ice-2023-08-22T21:28:21.787458591Z-17768.txt](https://github.com/rust-lang/rust/files/12413214/rustc-ice-2023-08-22T21.28.21.787458591Z-17768.txt)

### PR DESCRIPTION
**Pull request by notriddle**
Fixes #115064
Fixes #115062
Fixes #115116

### CODE REVIEW

File: src/librustdoc/passes/lint/redundant_explicit_links.rs [hunk 1]
```diff
@@ -98,13 +98,8 @@ fn check_redundant_explicit_link<'md>(
 
                 let explicit_link = dest.to_string();
                 let display_link = link_data.resolvable_link.clone()?;
-                let explicit_len = explicit_link.len();
```
Comment (ChAoSUnItY) [inline src/librustdoc/passes/lint/redundant_explicit_links.rs:101]: Same logic in [collect_infra_links.rs](https://github.com/rust-lang/rust/blob/fe5f5912579d02c4e6088c89574493dfa988531a/src/librustdoc/passes/collect_intra_doc_links.rs#L1435) should be replaced with new logic.

**Review by ChAoSUnItY — CHANGES_REQUESTED** [review]

**Review by ChAoSUnItY — APPROVED** [review]

**Comment by fmease** [comment]
This should also fix #115062, right?

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Nice thanks!

@bors r=GuillaumeGomez,ChAoSUnItY

**Comment by apiraino** [comment]
Beta backport approved

@rustbot label +beta-accepted

**Pull request merged**